api/envoy/api/v2/cds.proto

```diff
@@ -236,6 +236,82 @@ message Cluster {
   //   verification.
   auth.UpstreamTlsContext tls_context = 11;
 
+  // TransportSocketMatch specifies what transport socket config will be used
+  // when the matches conditions are satisfied.
+  message TransportSocketMatch {
+    // The name of the match.
+    string name = 1;
+
+    // Optional endpoint metadata match criteria.
+    // The connection to the endpoint with metadata matching what is set in this field
+    // will use the transport socket configuration specified here.
+    // The endpoints metadata entry in *envoy.transport_socket* is used to match
```

Comment:
@htuch @lizan 

I'm reading `well_known_names.h`, and see we already have `envoy.transport_sockets.xxx (alts)` for socket name, this makes me think using `envoy.transport_socket` to extract endpoint info can cause confusion.
https://github.com/envoyproxy/envoy/blob/master/source/extensions/transport_sockets/well_known_names.h#L17

how about renaming this to `envoy.transport_socket_match` or `envoy.transport_socket_selector`?

wdty?